### PR TITLE
Enhance Payload page and add OsLoader page

### PR DESCRIPTION
**Pull request by atharvalele**
- Enhance Payloads page in developer's guide
- Add OsLoader page to developer's guide

### CODE REVIEW

File: source/developer-guides/payload.rst [hunk 1]
```diff
@@ -28,11 +28,24 @@ Custom Payload
...
-  Implementation depends on |SPN| source code to build. It requires platform specific libraries.
+A payload can be **Tightly Coupled** or **Loosely Coupled** with Slim Bootloader depending on if the payload has **code/data dependency** or **data-only dependency** on the bootloader. \
 
-Loosely Coupled
-  Implementation can either be self-contained or depends on libraries in ``BootloaderCommonPkg`` and ``PayloadPkg``. The platform abstraction is provided through Hand-off-Blocks (HOB) data structures. These payloads are platform agnostic.
+* **Tightly Coupled**
+
+  * Implementation depends on |SPN| source code to build. It requires platform specific libraries.
+  * Can leverage code interfaces and libraries provided by the bootloader.
+  * |SPN| provides a set of EDK-2 based helper libraries like file system support, serial port debug message support, etc. to simplify the development of the payload.
```
Comment (rprangar) [inline source/developer-guides/payload.rst:37]: removed EDK-2

File: source/developer-guides/payload.rst [hunk 2]
```diff
@@ -28,11 +28,24 @@ Custom Payload
...
+  
+  Examples: OsLoader, FirmwareUpdate
+
+* **Loosely Coupled**
+
+  * A loosely coupled payload on the other hand, depends on platform initialization information passed by the bootloader to continue their initialization and boot process.
+  * Implementation can either be self-contained or depends on libraries in ``BootloaderCommonPkg`` and ``PayloadPkg``.
```
Comment (rprangar) [inline source/developer-guides/payload.rst:44]: Remove this - this is not valid "Implementation can either be self-contained or depends on libraries in ``BootloaderCommonPkg`` and ``PayloadPkg``."

File: source/developer-guides/payload.rst (hunk 1 above)
Comment (atharvalele) [reply source/developer-guides/payload.rst]: Removed, thanks

File: source/developer-guides/payload.rst (hunk 2 above)
Comment (atharvalele) [reply source/developer-guides/payload.rst]: Removed, thanks

**Review by rprangar — APPROVED** [review]
LGTM

**Pull request merged**